Edits to introducing pipelines section of AI Accelerator doc (#6496)
EnterpriseDB/docs: advocacy_docs/edb-postgres-ai/ai-accelerator/capabilities.mdx

---
title: "Capabilities"
navTitle: "Capabilities"

> description: "Capabilities of EDB Postgres AI - AI Accelerator Pipelines."

Author: Do we want to include EDB Postgres AI in the name of the product throughout this doc? It comes up many times and it seems clunky. but if EDB Postgres AI - AI Accelerator Pipelines is the official name, then I guess that's what we'd go with.

Reviewer: Thats what we are told we are need to do from now on.

---

> ## Pipeline lifecycle

Author: I was curious about the structure of this topic. It's called Capabilities but the one section is about the lifecycle. Will there by another section in the future? If not, I recommend changing the topic title to Pipeline lifecycle (or something more explicit), starting with introductory text (not a section head) and then popping everything up a level to level 2 heads. If there will be another section added in the future, it still would look better to start with introductory text rather than going right from a heading to a level 2 head, but I don't think we have any practice that says. not to do that.

Reviewer: More sections are expected as the capabilities grow. There's a lot of development happening here. Will discuss with the PM.

> This high-level overview describes the lifecycle of a pipeline in the Pipelines system.

Author: I was curious about the use of the word "system." I didn't get the sense of it as a system elsewhere. Just want to verify we want to refer to it as a system.

Reviewer: ```suggestion
This high-level overview describes the lifecycle of a pipeline within the AI Accellerator's Pipelines.
```

> ### Optionally, create a storage location

Author: I decided that active voice was indeed the way to go here. I think the topic title (lifecycle) indicates these are not instructions.

But another way to go, if you really don't want to use an imperative, would be:

Storage location creation (optional) 

Let's figure out what you want here and I will revise this accordingly if you don't like the change to active voice for the subheads.

Reviewer: This is describing a lifecycle, not participating in it. 

I'd need to look back at the whole doc, as this is based on other peoples descriptions.

Author: Okay, we'll circle back. I think the suggestion I made will work as "Storage location creation" would be part of the lifecycle and not indication participation in it.

This step is optional and needed only for accessing data external storage.

Data for processing can be stored in the database in a table or in an external storage location.

> If you want to use an external storage location, to access the data, you must create a storage location.

Author: Would this be clearer with "the" instead of "a"? 

...to access the data, you must create **the** storage location.

Reviewer: ```suggestion
If you want to use an external storage location to access data, you must create a storage location.
```

This storage location can be an S3 bucket or a local file system.

> The storage locations can be used to create a volume suitable for a retriever to use to access the data it contains.

Author: I found this sentence hard to parse. And should it be singular? (storage location vs storage locations) Does it mean:

In the storage location, you can create a volume for a retriever to use to access the data in the volume.

Reviewer: No...

```suggestion
The storage locations can be used by AI Accelerator to create a volume. This volume can then be used by a retriever to access its data.
```

> ### Create a model

Author: If you don't like the imperative, this could be changed to

Model creation

> Create a [model](models) with the Pipelines system. This model can be a machine learning model, a deep learning model, or any other type of model that can be used for AI tasks.

Author: ```suggestion
You create a [model](models) with the Pipelines system. This model can be a machine learning model, a deep learning model, or any other type of model that can be used for AI tasks.
```

> Create a [model](models) with the Pipelines system. This model can be a machine learning model, a deep learning model, or any other type of model that can be used for AI tasks.

Author: ```suggestion
Create a [model](models) with AI Accelerator Pipelines. This model can be a machine learning model, a deep learning model, or any other type of model that can be used for AI tasks.
```

### Create a retriever

> Create a retriever with the Pipelines system. A retriever is a function that retrieves data from a table or volume and returns it in a format that the model can use.

Author: ```suggestion
Create a retriever with AI Accelerator Pipelines. A retriever is a function that retrieves data from a table or volume and returns it in a format that the model can use.
```

* A name
* The name of a model to use

If the retriever is for a table, it also needs:

* The name of the source table
* The name of the column in the source table that contains the data
* The data type of the column

If the retriever is for a volume, it needs:

* The name of the volume
* The name of the column in the volume that contains the data

When you create a retriever, by default a vector table is created to store the embeddings of the data that's retrieved.
This table has a column to store the embeddings and a column to store the key of the data.

When you create the retriever, you can specify the name of the vector table and the name of the vector column and the key column. This ability is useful if you're migrating to aidb and want to use an existing vector table.

### Create embeddings

Embedding sees the data being retrieved from the source table or volume and encoded into a vector datatype. That vector data is then stored in the vector table.

If the source table already has data/rows when the retriever is created, then you need to make a manual *bulk embedding* call. This call generates the embeddings for all the existing data in the source table.

> You can then activate auto-embedding to keep the embeddings in sync going forward. Auto-embedding uses Postgres triggers to detect insertions and updates to the source table and generates embeddings for the new data.

Author: I did some research on how to spell auto-embedding. Auto on its own is not a word (unless you mean a car). It's a prefix in this context. So we could either hyphenate or not hyphenate (auto-embed or autoembed). The latter looks like a nonsense word, so I went with the hyphen. Elsewhere in this content, someone had used the hyphen, so now everything has the hyphen, which should be correct. (ChatGPT agrees!)

Reviewer: ChatGPT can go do one. 

But, yes, it should be hyphenated.
